### ISSUE TITLE
Wrong advice

### ISSUE THREAD

**Issue by zestful** [mdn/content#32140]
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/div

### What specific section or headline is this issue about?

Use of lang attribute

### What information was incorrect, unhelpful, or incomplete?

As a "pure" container, the <div> element does not inherently represent anything. Instead, it's used to group content so it can be easily styled using the [class](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes#class) or [id](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes#id) attributes, marking a section of a document as being written in a different [lang](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes#lang)uage (using the lang attribute), and so on.

### What did you expect to see?

Using a lang attribute on a div, will mean that a lot of assistive techs will ignore it, as it's not an element natively announced by AT's.   Kinda like having a label, and assigning it's for attribute to the wrapping div of an input field, instead of the input field itself.

I'd say this encourages developer bad practice giving this example, and causing problems for sites accessibility where people follow this advice

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/html/element/div`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/div
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/html/element/div/index.md
* Last commit: https://github.com/mdn/content/commit/e31cb5978e9f3c731c49db9ed0a15795b870e141
* Document last modified: 2023-10-14T00:23:21.000Z

</details>

**Comment by bsmth** [comment]
Hi @zestful - thanks for raising it. Just linking here for some notes: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/lang#accessibility_concerns

Just to clarify; you mean this is bad for AT:

```html
<div lang="de" class="lang-box">
  <p>Deutsch</p>
</div>
```

And this should be recommended instead?

```html
<div class="lang-box">
  <p lang="de">Deutsch</p>
</div>
```

**Comment by zestful** [comment]
> Hi @zestful - thanks for raising it. Just linking here for some notes: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/lang#accessibility_concerns
> 
> Just to clarify; you mean this is bad for AT:
> 
> ```
> <div lang="de" class="lang-box">
>   <p>Deutsch</p>
> </div>
> ```
> 
> And this should be recommended instead?
> 
> ```
> <div class="lang-box">
>   <p lang="de">Deutsch</p>
> </div>
> ```

Yes paragraph is a much better example to give thanks

**Comment by bsmth** [comment]
@zestful do you have any references that show this to be true? I cannot find any information to back up the recommendation.

**Comment by zestful** [comment]
@bsmth it's just basic HTML semantics, a div is a sectional element most commonly used for styling purposes not for content.

A div for example, will only be picked up if the user is 

a) Parsing the DOM an element at a time (very uncommon for people to do) & said div has parsable content either as inner HTML or a parsable attribute
b) A child element of a parent element currently being parsed by the user

Divs are more commonly user for presentation purposes or doing non-standard things compared to other elements, so there's a much larger chance that the AT won't be in the position of B above and just ignore it, likewise being a div some ATs can just ignore it all together unless in situation B & like said before said div has parseable content.  Also it introduces much greater scope for the developer to implement it wrong all together.

So it will have a much better chance of working as intended for users, if the recommendation when using the lang attribute as part of main page content, to suggest avoiding using it on div altogether.

**Comment by Josh-Cena** [comment]
Usage with `lang` is described in the [spec](https://html.spec.whatwg.org/multipage/grouping-content.html#the-div-element):

> It can be used with the [class](https://html.spec.whatwg.org/multipage/dom.html#classes), [lang](https://html.spec.whatwg.org/multipage/dom.html#attr-lang), and [title](https://html.spec.whatwg.org/multipage/dom.html#attr-title) attributes to mark up semantics common to a group of consecutive elements.

And an example is provided:

```html
<div lang="en-GB">
  <p>My other cat, coloured black and white, is a sweetie. He followed
  us to the pool today, walking down the pavement with us. Yesterday
  he apparently visited our neighbours. I wonder if he recognises that
  their flat is a mirror image of ours.</p>
  <p>Hm, I just noticed that in the last paragraph I used British
  English. But I'm supposed to write in American English. So I
  shouldn't say "pavement" or "flat" or "colour"...</p>
 </div>
```

If an AT doesn't announce it, it's a bug with said AT. The div doesn't need to be announced itself; it can just be used in replacement of marking individual children.
